### PR TITLE
feat(onboarding-1.5): Sidebar layout

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

TODO: WILL UPDATE INVITE TEAMMATE OPTION AND MAKE THINGS SLIGHTLY MORE MOBILE FRIENDLY IN ANOTHER PR

https://user-images.githubusercontent.com/25164963/169181182-74c8099e-5b4b-4aec-81ee-7d9441e03096.mov

Screen Shot 2022-05-18 at 9 14 45 PM
Screen Shot 2022-05-18 at 9 14 21 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by liyiy** [comment]
most recent cleanup:

Screen Shot 2022-05-19 at 12 55 17 PM

**Comment by liyiy** [comment]
note: The mocks aren't 100% finalized yet so this is just most of the functionality for things, will clean up most UI things in a follow up PR

**Review by clarkus — COMMENTED** [review]
Are there only three import plugins available so far? Are integrations also included as an import option?

**Comment by liyiy** [comment]
> Are there only three import plugins available so far? Are integrations also included as an import option?

there's 2 integrations (our two most popular ones) and 1 import plugin (the only one that exists with redshift) so far

File: frontend/src/scenes/ingestion/ingestionLogic.ts
Comment (rcmarron) [inline frontend/src/scenes/ingestion/ingestionLogic.ts:138]: I assume we want this to be behind a feature flag?

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (rcmarron) [inline frontend/src/scenes/ingestion/Sidebar.tsx:48]: There's some odd behavior when users click these buttons. For example, if you're on 'get started' and you click 'connect your product' it jumps to 'listen for events'. Then if you click 'get started' it moves to 'connect your product.

Because we require information from the previous step to inform what the next step shows, I wonder if these should not be clickable when moving forward?

Comment (rcmarron) [reply frontend/src/scenes/ingestion/Sidebar.tsx:48]: Oooo just seeing the `nonclickable` class, so I'm guessing this is just a bug. Also, it might makes sense to use `LemonButton` here instead of a click-able div.

File: frontend/src/layout/navigation/Navigation.tsx
Comment (rcmarron) [inline frontend/src/layout/navigation/Navigation.tsx:17]: Removing the top bar should probably be dependant on the same feature flag that adds the side bar. Otherwise, users in that side of the experiment can get trapped on the page.

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (rcmarron) [inline frontend/src/scenes/ingestion/Sidebar.tsx:58]: Most of the actions within the `SitePopover` don't work. For example, try clicking account settings, and the modal just closes.

What actions do we want to enable here?

Also, it seems like the ability to switch teams should be somewhere in here?

File: frontend/src/scenes/ingestion/CardContainer.tsx
Comment (rcmarron) [inline frontend/src/scenes/ingestion/CardContainer.tsx:31]: In Figam, it looks like the back button is a link above the container.

**Review by rcmarron — CHANGES_REQUESTED** [review]
Looking good! A few comments below.

Also, it seems like we should probably add a way for users to switch their teams on this page. Otherwise a user can accidentally create a new team and then get caught here.

File: frontend/src/scenes/ingestion/ingestionLogic.ts
Comment (liyiy) [reply frontend/src/scenes/ingestion/ingestionLogic.ts:138]: Yeah though I think it depends on how we want to implement the next experiment with 1.5 so I've been holding off on turning it into a feature flag just yet

File: frontend/src/layout/navigation/Navigation.tsx
Comment (liyiy) [reply frontend/src/layout/navigation/Navigation.tsx:17]: yes!! good catch, I totally missed that

File: frontend/src/scenes/ingestion/CardContainer.tsx
Comment (liyiy) [reply frontend/src/scenes/ingestion/CardContainer.tsx:31]: it looks unnoticeable and kind of out of place there :( 

Screen Shot 2022-05-20 at 12 48 41 PM

Comment (liyiy) [reply frontend/src/scenes/ingestion/CardContainer.tsx:31]: i'll put it above the container just to align with mocks

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (liyiy) [reply frontend/src/scenes/ingestion/Sidebar.tsx:58]: interesting, the buttons and actions are working for me?

Comment (rcmarron) [reply frontend/src/scenes/ingestion/Sidebar.tsx:58]: Hmm, for me, the URL changes, but then I immediately get redirected back to `/ingestion`. Seem the URL flip in the video below. I think it's because of the redirect logic in `sceneLogic`

https://user-images.githubusercontent.com/4813045/169576896-3aad0eb1-1c9e-403b-8fe8-a72746e8dea1.mov

`

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (liyiy) [reply frontend/src/scenes/ingestion/Sidebar.tsx:48]: I think that might've been the only bug with the nav steps? And yeah I'll replace these with lemon buttons for consistency

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (liyiy) [reply frontend/src/scenes/ingestion/Sidebar.tsx:58]: @clarkus What do you think about rick's comment here with having the ability to switch between projects during ingestion?

Comment (clarkus) [reply frontend/src/scenes/ingestion/Sidebar.tsx:58]: It's definitely a complicating factor. Is there a reason a user should be able to switch during ingestion? My concern would be that we don't prominently identify the project being affected by ingestion. If switching is necessary, we might need more prominent identification of the project and clear / obvious ways to switch. We could reuse the utility menu, but it's definitely not optimized for switching in this context.

Comment (clarkus) [reply frontend/src/scenes/ingestion/Sidebar.tsx:58]: If switching is important, it should probably be isolated to a specific point of the workflow. If you're mid-ingestion and you change projects, the snippets or code you use could change based on that switching. It could easily become unclear what values to use. This feels like something worth chatting about.

**Comment by lottiecoxon** [comment]
I only have one issue so far -the 'invite members' button is not inline with Chris's update (primary blue rectangle)

**Review by Twixes — DISMISSED** [review]
Pretty nice execution, but with some rough edges. ;)

File: frontend/src/scenes/ingestion/IngestionWizard.scss
Comment (neilkakkar) [reply frontend/src/scenes/ingestion/IngestionWizard.scss:56]: s/command/commend changes the tone completely 🙈

File: frontend/src/scenes/ingestion/IngestionWizard.scss
Comment (liyiy) [reply frontend/src/scenes/ingestion/IngestionWizard.scss:64]: can doo!

File: frontend/src/layout/navigation/TopBar/SitePopover.tsx
Comment (liyiy) [reply frontend/src/layout/navigation/TopBar/SitePopover.tsx:350]: tagged you in a figma comment regarding this

Comment (liyiy) [reply frontend/src/layout/navigation/TopBar/SitePopover.tsx]: i'm removing it from the sidebar and leaving it in the top right corner of the page for now. Will be making a follow up PR to address anything unresolved

File: frontend/src/scenes/ingestion/IngestionWizard.scss
Comment (liyiy) [reply frontend/src/scenes/ingestion/IngestionWizard.scss]: updated

**Comment by liyiy** [comment]
Current state of sidebar for now. Will make separate PRs for things left unresolved: email btn/modal fix, existing bug that rick found, and other UI display tweaks. Would like to get this PR merged in unless there's critical things that need addressing

Screen Shot 2022-05-23 at 1 06 58 PM

**Review by clarkus — DISMISSED** [review]
I think the sidebar needs more work. The alignment of elements is pretty inconsistent (I know it's like this in figma as well), but I think we should address this before merging. Ideally the branding is the same size as shown in the rest of the app with a similar placement. The redlines here illustrate how spacing is inconsistent from the left edge. 

The button also diverges from patterns we have elsewhere. When the button is filling its parent container, the label and icons should be center aligned, not left aligned as shown here. I don't think the border on the navigation is really necessary, but would defer to @lottiecoxon on that change. 

Screen Shot 2022-05-23 at 11 26 56 AM

**Comment by liyiy** [comment]
@clarkus these are lemon buttons though so if I totally align them to the left the button looks weird on hovering, just confirming if this is a desired effect? I am trying to re-use existing components we have here like lemon buttons. 

Screen Shot 2022-05-23 at 3 54 31 PM

**Comment by clarkus** [comment]
> @clarkus these are lemon buttons though so if I totally align them to the left the button looks weird on hovering, just confirming if this is a desired effect? I am trying to re-use existing components we have here like lemon buttons.

It should be from the edge of the element. Most of the feedback is with the size and placement of the logo. I think the placement should match what we use in the app header in terms of sizing and placement from the edge of the container. Not a blocker, just saying it looks off to me.

**Comment by liyiy** [comment]
@clarkus  OK i've updated the logo sizing and placement

Screen Shot 2022-05-23 at 4 20 40 PM

File: frontend/src/scenes/ingestion/Sidebar.tsx
Comment (liyiy) [reply frontend/src/scenes/ingestion/Sidebar.tsx]: OK maybe this is something to revisit then if we have extra time this week

**Comment by clarkus** [comment]
I changed the logo sizing and placement to align with our other brand examples in figma. Would like it if we can reuse the same sizing and offset here. This logo is 24px from the left and top of the container and uses the same sizing as that in the header. https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=10410%3A42692

**Comment by liyiy** [comment]
OK it's not 24 PX off to the left but that's because it's flex centered instead, if this alignment looks okay for now let's keep it at this and make any further improvements in a separate, final review instead

Screen Shot 2022-05-23 at 5 23 01 PM

**Comment by liyiy** [comment]
This isn't going to be shown to users yet so maybe we can re-align on the final look on things in another PR? @clarkus  @Twixes @rcmarron is there anything else that needs ASAP addressing?

File: frontend/src/scenes/ingestion/ingestionLogic.ts
Comment (alexkim205) [reply frontend/src/scenes/ingestion/ingestionLogic.ts:109]: What is the difference between currentIndex (reducer) and index (selector)? Can these concepts be merged or clarified with more specific naming?

**Review by alexkim205 — CHANGES_REQUESTED** [review]
Looking great and a good first step towards onboarding 1.5! Left a few minor inline comments.

Does disabling the third step if the current step is "Connect your product" make sense? This will keep the behavior consistent with when we're on the first step. 

Screen Shot 2022-05-24 at 11 21 18 AM

File: frontend/src/scenes/ingestion/ingestionLogic.ts
Comment (liyiy) [reply frontend/src/scenes/ingestion/ingestionLogic.ts:109]: index will eventually be removed it's just there because the old onboarding UI depends on it

File: frontend/src/scenes/ingestion/ingestionLogic.ts
Comment (liyiy) [reply frontend/src/scenes/ingestion/ingestionLogic.ts:138]: yep

**Comment by liyiy** [comment]
ONBOARDING 1.5 WITH TOPBAR AND SIDEBAR

FONT LOOKS LIGHTER ON BUTTONS THAT ARE DISABLED
Screen Shot 2022-05-25 at 5 01 35 PM

ONBOARDING 1.0 ONLY

Screen Shot 2022-05-25 at 5 09 12 PM

ORIGINAL ONBOARDING

Screen Shot 2022-05-25 at 5 08 42 PM

**Review by alexkim205 — APPROVED** [review]
looks like things work

**Review by rcmarron — APPROVED** [review]
SHIP IT!

**Pull request merged**
